### ISSUE TITLE
unimodules and RNN v1

### ISSUE THREAD

**Issue by pvinis** [unimodules/react-native-unimodules#71]
I saw that the docs have ways to fix AppDelegate.m for react-navigation and react-native-navigation v2, but what about v1? Unfortunately, we have an app with RNN v1 still, and we would like to add unimodules. Any ideas?

**Comment by pvinis** [comment]
Solved.

Apparently the line `self.moduleRegistryAdapter = [[UMModuleRegistryAdapter alloc] initWithModuleRegistryProvider:[[UMModuleRegistryProvider alloc] init]];` should happen before the bridge initialization. I guess the best way is to have this line as the first line in the applicationDidFinish.

**Comment by pvinis** [comment]
That didn't solve it completely. Now I have the modules initializing correctly, for example the keep-awake, but they are still not called. I get errors when something native should be called.

**Comment by pvinis** [comment]
I just get an undefined is not an object for ExpoKeepAwake :(

**Comment by sjchmiela** [comment]
Are you sure you've followed all the setup instructions (most specifically, adding this method to `AppDelegate`?)

```objc
- (NSArray<id<RCTBridgeModule>> *)extraModulesForBridge:(RCTBridge *)bridge
{
  NSArray<id<RCTBridgeModule>> *extraModules = [_moduleRegistryAdapter extraModulesForBridge:bridge];
  // You can inject any extra modules that you would like here, more information at:
  // https://facebook.github.io/react-native/docs/native-modules-ios.html#dependency-injection
  return extraModules;
}
```

**Comment by pvinis** [comment]
I did. I also put a breakpoint there and I could see the keepawake in there, with constants and a couple more.

**Comment by pvinis** [comment]
I'm on 0.59 with cocoapods if that helps.

**Comment by sjchmiela** [comment]
Hm, and what version of `react-native-unimodules` are you using? What is the output of `pod install` in `ios` directory?

**Comment by pvinis** [comment]
`0.7.0-rc.1`. The output is this:
```

Installing unimodules:
 expo-app-loader-provider@7.0.0 from ../node_modules/expo-app-loader-provider/ios
 expo-constants@7.0.0 from ../node_modules/expo-constants/ios
 expo-file-system@7.0.0 from ../node_modules/expo-file-system/ios
 expo-keep-awake@7.0.0 from ../node_modules/expo-keep-awake/ios
 expo-permissions@7.0.0 from ../node_modules/expo-permissions/ios
 unimodules-barcode-scanner-interface@4.0.0 from ../node_modules/unimodules-barcode-scanner-interface/ios
 unimodules-camera-interface@4.0.0 from ../node_modules/unimodules-camera-interface/ios
 unimodules-constants-interface@4.0.0 from ../node_modules/unimodules-constants-interface/ios
 unimodules-core@4.0.0 from ../node_modules/@unimodules/core/ios
 unimodules-face-detector-interface@4.0.0 from ../node_modules/unimodules-face-detector-interface/ios
 unimodules-file-system-interface@4.0.0 from ../node_modules/unimodules-file-system-interface/ios
 unimodules-font-interface@4.0.0 from ../node_modules/unimodules-font-interface/ios
 unimodules-image-loader-interface@4.0.0 from ../node_modules/unimodules-image-loader-interface/ios
 unimodules-permissions-interface@4.0.0 from ../node_modules/unimodules-permissions-interface/ios
 unimodules-react-native-adapter@5.0.0-alpha.0 from ../node_modules/@unimodules/react-native-adapter/ios
 unimodules-sensors-interface@4.0.0 from ../node_modules/unimodules-sensors-interface/ios
 unimodules-task-manager-interface@4.0.0 from ../node_modules/unimodules-task-manager-interface/ios

Installing unimodules:
 expo-app-loader-provider@7.0.0 from ../node_modules/expo-app-loader-provider/ios
 expo-constants@7.0.0 from ../node_modules/expo-constants/ios
 expo-file-system@7.0.0 from ../node_modules/expo-file-system/ios
 expo-keep-awake@7.0.0 from ../node_modules/expo-keep-awake/ios
 expo-permissions@7.0.0 from ../node_modules/expo-permissions/ios
 unimodules-barcode-scanner-interface@4.0.0 from ../node_modules/unimodules-barcode-scanner-interface/ios
 unimodules-camera-interface@4.0.0 from ../node_modules/unimodules-camera-interface/ios
 unimodules-constants-interface@4.0.0 from ../node_modules/unimodules-constants-interface/ios
 unimodules-core@4.0.0 from ../node_modules/@unimodules/core/ios
 unimodules-face-detector-interface@4.0.0 from ../node_modules/unimodules-face-detector-interface/ios
 unimodules-file-system-interface@4.0.0 from ../node_modules/unimodules-file-system-interface/ios
 unimodules-font-interface@4.0.0 from ../node_modules/unimodules-font-interface/ios
 unimodules-image-loader-interface@4.0.0 from ../node_modules/unimodules-image-loader-interface/ios
 unimodules-permissions-interface@4.0.0 from ../node_modules/unimodules-permissions-interface/ios
 unimodules-react-native-adapter@5.0.0-alpha.0 from ../node_modules/@unimodules/react-native-adapter/ios
 unimodules-sensors-interface@4.0.0 from ../node_modules/unimodules-sensors-interface/ios
 unimodules-task-manager-interface@4.0.0 from ../node_modules/unimodules-task-manager-interface/ios

Installing unimodules:
 expo-app-loader-provider@7.0.0 from ../node_modules/expo-app-loader-provider/ios
 expo-constants@7.0.0 from ../node_modules/expo-constants/ios
 expo-file-system@7.0.0 from ../node_modules/expo-file-system/ios
 expo-keep-awake@7.0.0 from ../node_modules/expo-keep-awake/ios
 expo-permissions@7.0.0 from ../node_modules/expo-permissions/ios
 unimodules-barcode-scanner-interface@4.0.0 from ../node_modules/unimodules-barcode-scanner-interface/ios
 unimodules-camera-interface@4.0.0 from ../node_modules/unimodules-camera-interface/ios
 unimodules-constants-interface@4.0.0 from ../node_modules/unimodules-constants-interface/ios
 unimodules-core@4.0.0 from ../node_modules/@unimodules/core/ios
 unimodules-face-detector-interface@4.0.0 from ../node_modules/unimodules-face-detector-interface/ios
 unimodules-file-system-interface@4.0.0 from ../node_modules/unimodules-file-system-interface/ios
 unimodules-font-interface@4.0.0 from ../node_modules/unimodules-font-interface/ios
 unimodules-image-loader-interface@4.0.0 from ../node_modules/unimodules-image-loader-interface/ios
 unimodules-permissions-interface@4.0.0 from ../node_modules/unimodules-permissions-interface/ios
 unimodules-react-native-adapter@5.0.0-alpha.0 from ../node_modules/@unimodules/react-native-adapter/ios
 unimodules-sensors-interface@4.0.0 from ../node_modules/unimodules-sensors-interface/ios
 unimodules-task-manager-interface@4.0.0 from ../node_modules/unimodules-task-manager-interface/ios

Installing unimodules:
 expo-app-loader-provider@7.0.0 from ../node_modules/expo-app-loader-provider/ios
 expo-constants@7.0.0 from ../node_modules/expo-constants/ios
 expo-file-system@7.0.0 from ../node_modules/expo-file-system/ios
 expo-keep-awake@7.0.0 from ../node_modules/expo-keep-awake/ios
 expo-permissions@7.0.0 from ../node_modules/expo-permissions/ios
 unimodules-barcode-scanner-interface@4.0.0 from ../node_modules/unimodules-barcode-scanner-interface/ios
 unimodules-camera-interface@4.0.0 from ../node_modules/unimodules-camera-interface/ios
 unimodules-constants-interface@4.0.0 from ../node_modules/unimodules-constants-interface/ios
 unimodules-core@4.0.0 from ../node_modules/@unimodules/core/ios
 unimodules-face-detector-interface@4.0.0 from ../node_modules/unimodules-face-detector-interface/ios
 unimodules-file-system-interface@4.0.0 from ../node_modules/unimodules-file-system-interface/ios
 unimodules-font-interface@4.0.0 from ../node_modules/unimodules-font-interface/ios
 unimodules-image-loader-interface@4.0.0 from ../node_modules/unimodules-image-loader-interface/ios
 unimodules-permissions-interface@4.0.0 from ../node_modules/unimodules-permissions-interface/ios
 unimodules-react-native-adapter@5.0.0-alpha.0 from ../node_modules/@unimodules/react-native-adapter/ios
 unimodules-sensors-interface@4.0.0 from ../node_modules/unimodules-sensors-interface/ios
 unimodules-task-manager-interface@4.0.0 from ../node_modules/unimodules-task-manager-interface/ios

Analyzing dependencies
Fetching podspec for `EXAppLoaderProvider` from `../node_modules/expo-app-loader-provider/ios`
Fetching podspec for `EXConstants` from `../node_modules/expo-constants/ios`
Fetching podspec for `EXFileSystem` from `../node_modules/expo-file-system/ios`
Fetching podspec for `EXKeepAwake` from `../node_modules/expo-keep-awake/ios`
Fetching podspec for `EXPermissions` from `../node_modules/expo-permissions/ios`
Fetching podspec for `RNAnalytics` from `../node_modules/@segment/analytics-react-native`
Fetching podspec for `RNCAsyncStorage` from `../node_modules/@react-native-community/async-storage`
Fetching podspec for `RNDeviceInfo` from `../node_modules/react-native-device-info`
Fetching podspec for `RNImageCropPicker` from `../node_modules/react-native-image-crop-picker`
Fetching podspec for `RNLocalize` from `../node_modules/react-native-localize`
Fetching podspec for `React` from `../node_modules/react-native`
Fetching podspec for `ReactNativePermissions` from `../node_modules/react-native-permissions`
Fetching podspec for `SentryReactNative` from `../node_modules/react-native-sentry`
Fetching podspec for `UMBarCodeScannerInterface` from `../node_modules/unimodules-barcode-scanner-interface/ios`
Fetching podspec for `UMCameraInterface` from `../node_modules/unimodules-camera-interface/ios`
Fetching podspec for `UMConstantsInterface` from `../node_modules/unimodules-constants-interface/ios`
Fetching podspec for `UMCore` from `../node_modules/@unimodules/core/ios`
Fetching podspec for `UMFaceDetectorInterface` from `../node_modules/unimodules-face-detector-interface/ios`
Fetching podspec for `UMFileSystemInterface` from `../node_modules/unimodules-file-system-interface/ios`
Fetching podspec for `UMFontInterface` from `../node_modules/unimodules-font-interface/ios`
Fetching podspec for `UMImageLoaderInterface` from `../node_modules/unimodules-image-loader-interface/ios`
Fetching podspec for `UMPermissionsInterface` from `../node_modules/unimodules-permissions-interface/ios`
Fetching podspec for `UMReactNativeAdapter` from `../node_modules/@unimodules/react-native-adapter/ios`
Fetching podspec for `UMSensorsInterface` from `../node_modules/unimodules-sensors-interface/ios`
Fetching podspec for `UMTaskManagerInterface` from `../node_modules/unimodules-task-manager-interface/ios`
Fetching podspec for `react-native-netinfo` from `../node_modules/@react-native-community/netinfo`
Fetching podspec for `react-native-shake` from `../node_modules/react-native-shake`
Fetching podspec for `react-native-slider` from `../node_modules/@react-native-community/slider`
Fetching podspec for `react-native-webview` from `../node_modules/react-native-webview`
Fetching podspec for `yoga` from `../node_modules/react-native/ReactCommon/yoga`
Downloading dependencies
Using Analytics (3.6.10)
Using Crashlytics (3.13.1)
Using EXAppLoaderProvider (7.0.0)
Using EXConstants (7.0.0)
Using EXFileSystem (7.0.0)
Using EXKeepAwake (7.0.0)
Using EXPermissions (7.0.0)
Using Fabric (1.10.1)
Using GruveoSDK (1.4)
Using GruveoWebRTC (1.65.25.2)
Using QBImagePickerController (3.4.0)
Using RNAnalytics (0.1.0-beta.0)
Using RNCAsyncStorage (1.5.1)
Using RNDeviceInfo (1.8.0)
Using RNImageCropPicker (0.21.3)
Using RNLocalize (1.2.0)
Using RSKImageCropper (2.2.1)
Using React (0.59.10)
Using ReactNativePermissions (1.1.1)
Using Sentry (4.1.3)
Using SentryReactNative (0.42.0)
Using UMBarCodeScannerInterface (4.0.0)
Using UMCameraInterface (4.0.0)
Using UMConstantsInterface (4.0.0)
Using UMCore (4.0.0)
Using UMFaceDetectorInterface (4.0.0)
Using UMFileSystemInterface (4.0.0)
Using UMFontInterface (4.0.0)
Using UMImageLoaderInterface (4.0.0)
Using UMPermissionsInterface (4.0.0)
Using UMReactNativeAdapter (5.0.0-alpha.0)
Using UMSensorsInterface (4.0.0)
Using UMTaskManagerInterface (4.0.0)
Using UrbanAirship-iOS-SDK (10.2.2)
Using react-native-netinfo (2.0.10)
Using react-native-shake (3.3.1)
Using react-native-slider (1.1.4)
Using react-native-webview (7.0.2)
Using yoga (0.59.10.React)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 44 dependencies from the Podfile and 39 total pods installed.

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`

[!] Unable to read the license file `LICENSE.md` for the spec `GruveoWebRTC (1.65.25.2)`
```

We have more targets, that's why the first part is repeated.

**Comment by pvinis** [comment]
Hm, I tried on a fresh 60 and it worked, and I tried on ours with RNNv1 on 60 and it didnt work. When I say it worked I mean the native method for activateKeepAwake for example got called, and the Constants were too. When I say it didn't work, I mean that the modules were initialized, appeared in the extraModules, but the native method was never called, and I get a red screen with ExpoKeepAwake.default.activate undefined is not an object.

**Comment by sjchmiela** [comment]
Interesting… could you try before you call any unimodules methods:
```js
import { NativeModules } from 'react-native';
console.log(NativeModules.NativeUnimoduleProxy);
```
?

**Comment by pvinis** [comment]
it prints `undefined`. hm but i just noticed this also printed: `The "UMNativeModulesProxy" native module is not exported through NativeModules; verify that @unimodules/react-native-adapter's native code is linked properly`  and `No native ExponentConstants module found, are you sure the expo-constants's module is linked properly?`

**Comment by pvinis** [comment]
Also, is there some slack or something that we could talk to solve this quicker? 🤔

**Comment by sjchmiela** [comment]
There's https://slack.expo.io, hit me up there @sjchmiela. 🙂

**Comment by pvinis** [comment]
done

**Comment by pvinis** [comment]
Solved for real this time, thanks to @sjchmiela!

The problem was that RNNv1 in the latest v1 on npm has only one init method, and it inits a new bridge. We already had one bridge, so that was creating a second one. All the unimodules were connected to the first bridge, and our app was then using the second bridge from RNN.

On github on the v1 branch there is a method `initWithDelegate` on RNN, so I just copied that and used `patch-package` to keep it on our code, and then used that function to start RNN. Now everything works as expected.

The diff for that line in my AppDelegate is:
```
- [[RCCManager sharedInstance] initBridgeWithBundleURL:jsCodeLocation launchOptions:launchOptions];
+ [[RCCManager sharedInstance] initBridgeWithDelegate:self launchOptions:launchOptions];
```
